hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -126,7 +127,7 @@ public void commit(TableMetadata base, TableMetadata metadata) {
       // TODO add lock heart beating for cases where default lock timeout is too low.
       Table tbl;
       if (base != null) {
-        tbl = metaStoreClient.get_table(database, tableName);
+        tbl = metaClients.run(client -> client.getTable(database, tableName));
```

Comment:
One question that is not related to this change. Am I correct that we don't refresh the metadata after we require the lock?

Reply:
This line is after the lock is acquired on line 125.

Reply:
Correct, but the refresh happens on line 111, right? Looks like we don't refresh after acquiring the lock, which means the table metadata can be changed after the refresh but before we acquire the lock?

Reply:
Line 111 (`base != current()`) only checks the currently-known metadata. If before a refresh the base metadata is already out of date, it fails early.

After the lock is acquired, table metadata is refreshed using this call to `getTable` on line 130 (in the latest version). On line 148, this checks the metadata location that was fetched to ensure that it is the one that was expected.

Reply:
You are right, we handle it correctly.